go/vt/vtadmin/cluster/discovery/fakediscovery/discovery.go

```diff
@@ -103,19 +115,28 @@ func (d *Fake) AddTaggedVtctlds(tags []string, vtctlds ...*vtadminpb.Vtctld) {
 // SetGatesError instructs whether the fake should return an error on gate
 // discovery functions.
 func (d *Fake) SetGatesError(shouldErr bool) {
+	d.m.Lock()
+	defer d.m.Unlock()
+
 	d.gates.shouldErr = shouldErr
 }
 
 // SetVtctldsError instructs whether the fake should return an error on vtctld
 // discovery functions.
 func (d *Fake) SetVtctldsError(shouldErr bool) {
+	d.m.Lock()
+	defer d.m.Unlock()
+
 	d.vtctlds.shouldErr = shouldErr
 }
 
 var _ discovery.Discovery = (*Fake)(nil)
 
 // DiscoverVTGates is part of the discovery.Discovery interface.
 func (d *Fake) DiscoverVTGates(ctx context.Context, tags []string) ([]*vtadminpb.VTGate, error) {
+	d.m.Lock()
```

Comment:
note that all the other DiscoverVTGate* functions call this, so they get the lock for free (ditto on vtctld functions)